app/src/main/java/com/example/marvelapp/service/request/generator/MarvelRequestGenerator.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.marvelapp.service.request.generator
+
+import okhttp3.OkHttpClient
+import org.koin.core.component.KoinComponent
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class MarvelRequestGenerator : KoinComponent {
+    private val httpClient = OkHttpClient.Builder().addInterceptor { chain ->
+        val defaultRequest = chain.request()
+        val defaultHttpUrl = defaultRequest.url()
+        val httpUrl = defaultHttpUrl.newBuilder()
+            .addQueryParameter(TS, TS_VALUE)
+            .addQueryParameter(API, API_VALUE)
+            .addQueryParameter(HASH, HASH_VALUE)
+            .build()
+        val requestBuilder = defaultRequest.newBuilder().url(httpUrl)
+        chain.proceed(requestBuilder.build())
+    }
+
+    private val builder = Retrofit.Builder()
+        .baseUrl(MARVEL_BASE_URL)
+        .addConverterFactory(GsonConverterFactory.create())
+
+    fun <S> createService(serviceClass: Class<S>): S {
+        val retrofit = builder.client(httpClient.build()).build()
+        return retrofit.create(serviceClass)
+    }
+
+    companion object {
+        private const val MARVEL_BASE_URL = "https://gateway.marvel.com"
+        private const val TS = "ts"
+        private const val API = "apikey"
+        private const val HASH = "hash"
```

Comment:
MARVEL_BASE_URL, API_VALUE, and HASH_VALUE should be declared in Gradle as a "buildConfigField"

Reply:
fixed